osu.Game/Tests/Visual/OnlinePlay/BasicTestRoomManager.cs

```diff
@@ -33,12 +33,18 @@ public void CreateRoom(Room room, Action<Room> onSuccess = null, Action<string>
             room.RoomID.Value ??= Rooms.Select(r => r.RoomID.Value).Where(id => id != null).Select(id => id.Value).DefaultIfEmpty().Max() + 1;
             onSuccess?.Invoke(room);
 
-            AddRoom(room);
+            AddOrUpdateRoom(room);
         }
 
-        public void AddRoom(Room room)
+        public void AddOrUpdateRoom(Room room)
```

Comment:
This indirectly breaks `AddRooms()`, because that method always uses room IDs of `0..(count - 1)`, which in turn breaks `TestSceneLoungeRoomsContainer.TestRulesetFiltering` (the catch rooms added therein overwrite the osu! rooms).